`quint-examples-dashboard` CI job fails on macOS

The `quint-examples-dashboard` CI job fails on macOS, due to missing GNU parallel.
This never ran, because we didn't touch examples since the it was added, and the make target wasn't phony before #1003.

```
Run # Update the examples dashboard
echo "<!-- DO NOT EDIT: THIS FILE IS GENERATED FROM examples/.scripts/README-text.md VIA 'make examples' -->" > examples/README.md
cat examples/.scripts/README-text.md >> examples/README.md
./examples/.scripts/run-examples.sh >> examples/README.md
./examples/.scripts/run-examples.sh: line 24: parallel: command not found
make: *** [examples/README.md] Error 127
Error: Process completed with exit code 2.
```

Example job log:
https://github.com/informalsystems/quint/actions/runs/5444694387/jobs/9919931065#logs

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -142,6 +142,11 @@ jobs:
           make apalache
         env:
           GH_TOKEN: ${{ github.token }}
+      - if: matrix.operating-system == 'macos-latest'
+        # TODO(#1004): Workaround for GNU parallel not being available on macOS
+        # We should find a better way of managing dev dependencies.
+        name: Install GNU parallel
+        run: brew install parallel
       - name: Ensure the examples dashboard is up to date
         run: |
           # Update the examples dashboard

